src/clientandserver/FactorServiceImpl.java

```diff
@@ -1,95 +1,85 @@
 package com.example.clientandserver;
 
-import java.util.List;
-import java.util.stream.Collectors;
 
 import apiProto.CoordinatorEngine.coordinatorResponse;
-import apiProto.DataStore.DataStoreReadRequest;
-import apiProto.DataStore.DataStoreReadResponse;
-import apiProto.DataStore.DataStoreWriteRequest;
+
 // import apiProto.DataStoreServiceGrpc;
 import apiProto.DataStoreServiceGrpc.DataStoreServiceBlockingStub;
 import apiProto.FactorServiceGrpc.FactorServiceImplBase;
 import com.example.factorfinder.FactorComputationCoordinatorImpl;
 import com.example.factorfinder.FactorComputeEngineImpl;
 import com.example.factorfinder.FactorComputeRequest;
+import com.example.factorfinder.FactorComputeResult;
 //import factorFinder.FactorComputeResult;
-import com.example.factorfinder.FactorDataStoreImpl;
 import com.example.factorfinder.FileInputConfig;
 import com.example.factorfinder.FileOutputConfig;
 
 public class FactorServiceImpl extends FactorServiceImplBase {
   private FactorComputationCoordinatorImpl coordinator;
-  private DataStoreServiceBlockingStub dataStoreStub;
-
+  public DataStoreServiceBlockingStub dataStoreStub;
   // Constructor
-
   public FactorServiceImpl(FactorComputeEngineImpl computeEngine,
-      FactorDataStoreImpl dataStore,
+
       DataStoreServiceBlockingStub dataStoreStub) {
     this.dataStoreStub = dataStoreStub;
     this.coordinator =
-        new FactorComputationCoordinatorImpl(computeEngine, dataStore);
+        new FactorComputationCoordinatorImpl(computeEngine, dataStoreStub);
   }
-
   public void compute(apiProto.CoordinatorEngine.coordinatorRequest request,
       io.grpc.stub.StreamObserver<
           apiProto.CoordinatorEngine.coordinatorResponse> responseObserver) {
     // create a request from API 1 and save it
-    FileInputConfig internallInputFile =
+    FileInputConfig internalInputFile =
         new FileInputConfig(request.getInputFile());
     FileOutputConfig internalOutputFile =
         new FileOutputConfig(request.getOutputFile());
     char internalDelimier = request.getDelimiter().charAt(0);
-
     FactorComputeRequest internalRequest = new FactorComputeRequest(
-        internallInputFile, internalOutputFile, internalDelimier);
+        internalInputFile, internalOutputFile, internalDelimier);
 
     // Save result from request into the internal type
     com.example.factorfinder.FactorComputeResult internalResponse = coordinator.compute(
         internalRequest); // TODO : make type FactorComputeRequest and get
                           // Result but need to make it back into the
                           // coordinatorRepsonse
```

Comment:
I tested everything in eclipse. Everything works, great work.

Reply:
ok so can you approve, so i can merge it to main